### PR TITLE
Add resource definition refactor

### PR DESCRIPTION
**Pull request by gravesm**
##### SUMMARY

This refactors most of the logic around creating a list of functional
resource definitions based on input parameters for the module.

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by alinabuzachis — APPROVED** [review]
LGTM!

File: plugins/module_utils/k8s/resource.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import os
```
Comment (Akasurde) [inline plugins/module_utils/k8s/resource.py:1]: Can we add a license boilerplate here? and a changelog entry for this change.

Comment (gravesm) [reply plugins/module_utils/k8s/resource.py:1]: License definitely. My thinking is to create a single changelog entry for the refactor when we merge this branch into main.

File: plugins/module_utils/k8s/resource.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (Akasurde) [inline plugins/module_utils/k8s/resource.py:16]:
```suggestion
# Copyright: (c) 2021, Red Hat | Ansible
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```

**Comment by gravesm** [comment]
recheck

**Pull request merged**